examples/covariance/plot_outlier_detection.py

```diff
@@ -72,19 +79,23 @@
     X = np.r_[X, np.random.uniform(low=-6, high=6, size=(n_outliers, 2))]
 
     # Fit the model
-    plt.figure(figsize=(10.8, 3.6))
+    plt.figure(figsize=(9, 7))
     for i, (clf_name, clf) in enumerate(classifiers.items()):
         # fit the data and tag outliers
-        clf.fit(X)
-        scores_pred = clf.decision_function(X)
+        if clf_name == "Local Outlier Factor":
```

Comment:
This is not a good sign. So this is because predicting on the training set doesn't work for ``KNeighbors``... I guess we discussed this before. It's still not great :( I feel this is more of an issue here than in the neighbors module because neighbors are usually used in a supervised setting and this is used without a train/test split... gah...
Anyhow, I know people will use it wrongly... but I don't know how to fix it...